src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -942,6 +952,16 @@ public BoundExpression Typeof(TypeSymbol type)
             { WasCompilerGenerated = true };
         }
 
+        public BoundExpression TypeofBeforeRewriting(TypeSymbol type)
+        {
+            return new BoundTypeOfOperator(
+                Syntax,
+                Type(type),
+                // For some reason the rewriting logic asserts that the GetTypeFromHandle property is null.
+                null,
```

Comment:
Guess: We use the same representation before and after lowering and the presence or absence of that field is used to distinguish the two cases.